packages/table/src/interactions/selectable.tsx

```diff
@@ -118,33 +127,43 @@ export class DragSelectable extends React.Component<IDragSelectableProps, {}> {
             return false;
         }
 
-        const focusCellCoordinates = DragSelectable.getFocusCellCoordinatesFromRegion(region);
+        const {
+            ignoreSelectedRegionClicks,
+            onSelection,
+            selectedRegions,
+            selectedRegionTransform,
+        } = this.props;
 
+        const focusCellCoordinates = DragSelectable.getFocusCellCoordinatesFromRegion(region);
         this.props.onFocus(focusCellCoordinates);
 
-        if (this.props.selectedRegionTransform != null) {
-            region = this.props.selectedRegionTransform(region, event);
+        if (selectedRegionTransform != null) {
+            region = selectedRegionTransform(region, event);
         }
 
-        const foundIndex = Regions.findMatchingRegion(this.props.selectedRegions, region);
+        if (ignoreSelectedRegionClicks && Regions.containsRegion(selectedRegions, region)) {
+            return false;
+        }
+
+        const foundIndex = Regions.findMatchingRegion(selectedRegions, region);
         if (foundIndex !== -1) {
-            // If re-clicking on an existing region, we either carefully
-            // remove it if the meta key is used or otherwise we clear the
-            // selection entirely.
             if (DragEvents.isAdditive(event)) {
-                const newSelectedRegions = this.props.selectedRegions.slice();
+                // If re-clicking on an existing region, we either carefully
+                // remove it if the meta key is used or otherwise we clear the
+                // selection entirely.
```

Comment:
i think this comment belongs before the `if`, not after (cuz it also takes about the `else`)